src/main/java/fi/aalto/cs/apluscourses/model/Submission.java

```diff
@@ -0,0 +1,55 @@
+package fi.aalto.cs.apluscourses.model;
+
+import java.nio.file.Path;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public class Submission {
+
+  @NotNull
+  private final Exercise exercise;
+  @NotNull
+  private final SubmissionInfo submissionInfo;
+  @NotNull
+  private final Map<String, Path> files;
+  @NotNull
+  private final Group group;
+
+  /**
+   * Constructs a new object instance.
+   *
+   * @param exercise       Exercise.
+   * @param submissionInfo Information for the submission.
+   * @param files          Map from keys to file paths.
+   * @param group          Group in which the submission is made.
+   */
+  public Submission(@NotNull Exercise exercise,
+                    @NotNull SubmissionInfo submissionInfo,
+                    @NotNull Map<String, Path> files,
+                    @NotNull Group group) {
```

Comment:
Nit: the comment here could mention what the keys in the map are

Reply:
Right. Now it's just stupid. 😅